src/sys/unix.rs

```diff
@@ -1248,21 +1243,26 @@ fn timeval2dur(raw: libc::timeval) -> Option<Duration> {
     }
 }
 
-fn to_s_addr(addr: &Ipv4Addr) -> libc::in_addr_t {
-    let octets = addr.octets();
-    u32::from_ne_bytes(octets)
+pub(crate) fn to_in_addr(addr: &Ipv4Addr) -> in_addr {
+    in_addr {
+        s_addr: u32::from_ne_bytes(addr.octets()),
+    }
 }
 
-fn from_s_addr(in_addr: libc::in_addr_t) -> Ipv4Addr {
-    in_addr.to_be().into()
+pub(crate) fn from_in_addr(in_addr: in_addr) -> Ipv4Addr {
+    Ipv4Addr::from(in_addr.s_addr.to_ne_bytes())
```

Comment:
These helper methods were really handy. Removed a lot of code from the `sockaddr.rs` file and got rid of a number of `#[cfg(...)]`s. So I made them `pub(crate)` so I could use them in the other module. I alos changed the abstraction one level. So instead of converting to/from the only field in the `in_addr`/`in6_addr` I made these helpers convert directly into that struct. This in turn helped remove some repeated code here in the two sys files as well.